cve_bin_tool/extractor.py

```diff
@@ -208,11 +201,10 @@ def __enter__(self):
         return self
 
     def __exit__(self, exc_type, exc_val, exc_tb):
-        shutil.rmtree(self.tempdir)
+        with ErrorHandler(mode=self.error_mode, logger=self.logger):
+            shutil.rmtree(self.tempdir)
 
 
-class Extractor(BaseExtractor):
+def Extractor(*args, **kwargs):
     """Provides a context which extraction is done in"""
-
-    def __call__(self, *args, **kwargs):
-        return TempDirExtractorContext(*args, **kwargs)
+    return TempDirExtractorContext(*args, **kwargs)
```

Comment:
Create function instead of callable class for better readability. Ex:
`Extractor()(error_mode=mode)` will now become `Extractor(error_mode=mode)`